omod/src/main/java/org/openmrs/module/queue/web/QueueEntryTransitionRestController.java

```diff
@@ -96,6 +101,36 @@ public Object transitionQueueEntry(@RequestBody QueueEntryTransitionRequest body
 		
 		transition.setNewPriorityComment(body.getNewPriorityComment());
 		
+		// Location waiting for
+		if (rawBody.containsKey(QueueEntryTransitionRequest.NEW_LOCATION_WAITING_FOR_FIELD)) {
+			if (body.getNewLocationWaitingFor() == null) {
+				transition.setNewLocationWaitingFor(null);
+			} else {
+				Location location = services.getLocationService().getLocationByUuid(body.getNewLocationWaitingFor());
+				if (location == null) {
+					throw new APIException("Invalid locationWaitingFor specified: " + body.getNewLocationWaitingFor());
+				}
+				transition.setNewLocationWaitingFor(location);
+			}
+		} else {
+			transition.setNewLocationWaitingFor(queueEntry.getLocationWaitingFor());
+		}
+		
+		// Provider waiting for
+		if (rawBody.containsKey(QueueEntryTransitionRequest.NEW_PROVIDER_WAITING_FOR_FIELD)) {
+			if (body.getNewProviderWaitingFor() == null) {
+				transition.setNewProviderWaitingFor(null);
+			} else {
+				Provider provider = services.getProviderService().getProviderByUuid(body.getNewLocationWaitingFor());
```

Comment:
This is a bug, probably related to copying the code block.  This should be `body.getNewProviderWaitingFor()`.  I'm surprised the tests didn't catch this - can you modify the tests to ensure they do?